sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs

```diff
@@ -229,7 +229,7 @@ public virtual Response Upload<T>(string ruleId, string streamName, IEnumerable<
 
                     if (response.Status != 204)
                     {
-                        throw new RequestFailedException(response);
+                        OnException(batch.LogsCount, options, response, cancellationToken);
```

Comment:
Do we need to clone the options so that users don't change the event handler underneath us?

Reply:
In the RaiseAsync code, a reference to the EventHandler is used so it can't be modified then I believe

Reply:
We wouldn't observe any modifications after we call RaiseAsync, but I thought we wanted to avoid having the user modify it out from under us after they call the Upload method.

Reply:
I could go either way on this one.  Always referencing a fresh copy means people can do things like remove their handler if they decide they want to stop listening after the first time the event is raised.  The race condition situation where we assign it to a local is more about avoiding a very easy to make NullRefEx.

Reply:
Based on the Teams clarification with Ted, I think we can remove the clone method. We can instead just do something like:
```c#
options ??= new UploadOptions();
```
in the top of Upload.

Reply:
If options is null and we instantiate new Options(), I don't think I would know when the EventHandler is enabled. The code currently uses null checks on Options to determine if there is an EventHandler. Could we not modify the existing options and use it as is?

Reply:
That's fair. I guess we could only do a null check for handler within the options class itself, so maybe not worth it.